proto/buf/alpha/registry/v1alpha1/module.proto

```diff
@@ -32,13 +32,11 @@ message LocalModuleReference {
 //
 // It does not include a remote.
 message LocalModulePin {
-  reserved 5;
-  reserved "digest";
-
   string owner = 1;
   string repository = 2;
   string branch = 3;
   string commit = 4;
+  string digest = 5;
```

Comment:
This is usually an anti-pattern in proto, to try to re-use reserved fields. What was it previously used for? Are we certain there will be no serialized protos around that might be using the old field and get incorrectly interpreted as the new one? If that did happen, what is the consequence?

Reply:
> What was it previously used for?

It was used for digests too: b1 and b3 digests. They had "b1" and "b3" prefixes in their string form.

> Are we certain there will be no serialized protos around that might be using the old field and get incorrectly interpreted as the new one?

It gets better: are we certain there are/will-be no serialized YAML forms given these structs are mapped through proto form into/from `buf.lock`.

Answer: they can exist in pre-v1.0 releases modules. Even so, these values cannot be interpreted in our new digest form.

> If that did happen, what is the consequence?

The module with a b1/3 digest in their `buf.lock` would fail to build.

> This is usually an anti-pattern in proto, to try to re-use reserved fields.

A different field number would add wire level protection.

Reply:
Updated to keep existing field reserved (in case they are stored on disk) and add a new field for the module digest.